homeworks/nagynorbert/3_conditionals_loops/rock_paper_scissors.py

```diff
@@ -0,0 +1,70 @@
+# homework - Nagy Norbert
+
+# define numbers of rounds
+while True:
+    rounds = int(input("Please define how many rounds would you like to play: "))
+    if rounds < 3 or rounds%2 == 0:
+        print("Invalid input, please define a positive odd number which grater than 2")
+    else:
+        print("Thanks, valid input.")
+        break
+
+valid_signs = ["rock","paper","scissors"]
+
+# define and validate signs in one round
+def add_input_sign(player):
+    print(f" -  {player}.player  - ")
+    while True:
+        sign = input("Please add your sign: ")
+        input_sign = sign.strip().lower()
+        if sign not in valid_signs:
+            print("Invalid input: {sign}, try again")
+        else:
+            print(f"Thanks, valid input: {sign}.")
+            break
+
+    return input_sign
+
+# evaluate signs and summerize points in one round
+def evaluate_signs(sum_points_1,sum_points_2,player_1_sign,player_2_sign):
```

Comment:
elég hosszú így minden lehetséges kimenetelt megvizsgálni, ebben a formában talán egyszerűbb ha van egy ún. mapping-et a győztes kombinációkról, és ezeket vizsgálod:

```
def evaluate_signs(sum_points_1, sum_points_2, player_1_sign, player_2_sign):

    winning_combinations = {
        "rock": "scissors",
        "scissors": "paper",
        "paper": "rock"
    }
    
    if player_1_sign == player_2_sign:
        return sum_points_1, sum_points_2
    elif winning_combinations[player_1_sign] == player_2_sign:
        return sum_points_1 + 1, sum_points_2
    else:
        return sum_points_1, sum_points_2 + 1
```

Illetve szuper lenne ha kapnánk minden körben is visszajelzést arról, hogy az adott kört ki nyerte, és jelenleg mi az állás.